app/models.py

```diff
@@ -330,48 +330,22 @@ class SupplierFramework(db.Model):
-    # This framework_agreements table is now the source of truth for this data
-    agreement_returned_at = db.Column(db.DateTime, index=False, unique=False, nullable=True)
-    countersigned_at = db.Column(db.DateTime, index=False, unique=False, nullable=True)
-    agreement_details = db.Column(JSON)
-
     supplier = db.relationship(Supplier, lazy='joined', innerjoin=True)
     framework = db.relationship(Framework, lazy='joined', innerjoin=True)
 
-    @property
-    def current_framework_agreement(self):
-        """
-        For the moment, we include drafts in the SupplierFramework to keep our interface the same whilst we refactor the
-        frontend to allow multiple framework agreements per supplier framework. This means that if a draft is created
-        after any other framework agreement then it must take precident (as the supplier frontend for the moment only
-        knows about one framework agreement (the current framework agreement) we must expose the draft so it can be
-        edited as they complete the signing flow). Note, as we have no timestamp for framework agreements being created
-        we instead have to use the highest `id`. Also note, this means that for a short while, if someone has
-        signed/countersigned a supplier framework and then starts a new draft then the supplier framework will return to
-        'draft' status and will lose knowledge of the previous signing.
-
-        After the supplier frontend has been refactored so that the signing flow deals with a framework agreement rather
-        than a suqpplier framework then we will be able to ignore drafts and not include them in the current framework
-        agreement
-        """
-        if self.framework_agreements:
-            if self.framework_agreements[-1].status == 'draft':
-                return self.framework_agreements[-1]
-
-            most_recently_signed_or_countersigned = self.framework_agreements[-1]
-            most_recent_time = most_recently_signed_or_countersigned.most_recent_signature_time
-
-            for fa in self.framework_agreements:
-                if fa.status == "draft":
-                    continue
-
-                if fa.most_recent_signature_time > most_recent_time:
-                    most_recently_signed_or_countersigned = fa
-                    most_recent_time = fa.most_recent_signature_time
-
-            return most_recently_signed_or_countersigned
+    current_framework_agreement = db.relationship(
+        lambda: FrameworkAgreement,
+        lazy="subquery",
+        order_by=lambda: desc(FrameworkAgreement.most_recent_signature_time),
+        primaryjoin=lambda: and_(
+            SupplierFramework.supplier_id == FrameworkAgreement.supplier_id,
+            SupplierFramework.framework_id == FrameworkAgreement.framework_id,
+            FrameworkAgreement.status != "draft"
+        ),
+        distinct_target_key=True,
+        uselist=False,
+        viewonly=True,
+    )
```

Comment:
Hmmm... come to think of it do we need to specify that this should be an outer join? How else would it be able to return a row for a `SupplierFramework` that had no `FrameworkAgreement`s?